OpenSim/Examples/BuildDynamicWalker/BuildDynamicWalkerModel.cpp

```diff
@@ -0,0 +1,352 @@
+        double mass = 1;
+
+        // Location of the center of mass from the body origin expressed in the
+        // body frame.
+        Vec3 comLocInBody(0.0, 0.0, 0.0);
+
+        // Inertia of the body expressed in the body frame
+        Inertia bodyInertia(1.0, 1.0, 1.0, 0.0, 0.0, 0.0);
+
+        // Create the body
+        Body* platform = new Body("Platform", mass, comLocInBody, bodyInertia);
+
+        // Add geometry to display in the GUI
+        platform->attachGeometry(new Brick(Vec3(1, 0.05, 1)));
+
+        // Add the Platform Body to the Model
+        osimModel.addBody(platform);
+
+        // Section: Create the Platform Joint
+        // Create the joint connection the platform to the ground
+        Vec3 locationInParent(0.0, 0.0, 0.0);
+        Vec3 orientationInParent(0.0, 0.0, 0.0);
+        Vec3 locationInChild(0.0, 0.0, 0.0);
+        Vec3 orientationInChild(0.0, 0.0, 0.0);
+        PinJoint *platformToGround = new PinJoint("PlatformToGround",
+            ground, locationInParent, orientationInParent,
+            *platform, locationInChild, orientationInChild);
+
+        // Section: Set the properties of the coordinates that define the joint
+        // A pin joint consists of a single coordinate describing a change in
+        // orientation about the Z axis
+        Coordinate& platform_rz =
+            platformToGround->updCoordinate(PinJoint::Coord::RotationZ);
+        platform_rz.setName("platform_rz");
+        double platform_rz_range[2] = {-Pi / 2.0, 0};
+        platform_rz.setRange(platform_rz_range);
+        platform_rz.setDefaultValue(convertDegreesToRadians(-10.0));
+        platform_rz.setDefaultLocked(true);
+
+        // Add the PlatformToGround Joint to the Model
+        osimModel.addJoint(platformToGround);
+
+        // Section: Create the Pelvis
+        mass = 1;
+
+        // Location of the Center of Mass from the Body Origin expressed in
+        // Body Frame
+        comLocInBody = Vec3(0.0, 0.0, 0.0);
+
+        // Inertia of the Body expressed in the Body Frame
+        bodyInertia = Inertia(1.0, 1.0, 1.0, 0.0, 0.0, 0.0);
+
+        // Create the body
+        Body* pelvis = new Body("Pelvis", mass, comLocInBody, bodyInertia);
+
+        // Add geometry to display in the GUI
+        pelvis->attachGeometry(new Ellipsoid(
+                    pelvisWidth / 4.0, pelvisWidth / 4.0, pelvisWidth / 2.0));
+
+        // Add the Pelvis Body to the Model
+        osimModel.addBody(pelvis);
+
+        // Create the joint which connects the Pelvis to the Platform
+        locationInParent = Vec3(0.0, 0.0, 0.0);
+        orientationInParent = Vec3(0.0, 0.0, 0.0);
+        locationInChild = Vec3(0.0, 0.0, 0.0);
+        orientationInChild = Vec3(0.0, 0.0, 0.0);
+        PlanarJoint *pelvisToPlatform = new PlanarJoint("PelvisToPlatform",
+            *platform, locationInParent, orientationInParent,
+            *pelvis, locationInChild, orientationInChild);
```

Comment:
In the interest of streamlining and avoiding the creation of empty transforms, you could use this `Joint` constructor instead:
```cpp
Joint(const std::string&    name,
          const PhysicalFrame&  parent,
          const PhysicalFrame&  child);
```

I think it would look a lot cleaner.